test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -2015,4 +2052,17 @@ protected FieldCapabilitiesResponse fieldCaps(
             return FieldCapabilitiesResponse.fromXContent(parser);
         }
     }
+
+    protected static XContentParser toParser(Response response) throws IOException {
+        return XContentHelper.createParser(XContentParserConfiguration.EMPTY, toBytes(response), XContentType.JSON);
+    }
+
+    protected static BytesReference toBytes(Response response) throws IOException {
+        return new BytesArray(EntityUtils.toByteArray(response.getEntity()));
+    }
+
+    protected static Map<?, ?> toMap(Response response) throws IOException {
+        return XContentHelper.convertToMap(XContentType.JSON.xContent(), response.getEntity().getContent(), false);
+    }
```

Comment:
Is it necessary to have the `Map<?, ?>` response for this? We already have the `responseAsMap` method in `ESRestTestCase` that returns a `Map<String, Object>`, do we need both?